source/Sagan-RDBMS-Tests/PooledRDBMSRepositoryProviderTest.class.st

```diff
@@ -97,6 +139,394 @@ PooledRDBMSRepositoryProviderTest >> tearDown [
+
+	| sdf1 updatedSDF1 vf1A |
+
+	vf1A := Spaceship named: 'VF-1A'.
+	sdf1 := Mothership named: 'SDF-1' withAll: {}.
+	updatedSDF1 := Mothership named: 'SDF-1*' withAll: {vf1A}.
+	ships store: sdf1.
+	ships update: sdf1 with: updatedSDF1.
+	self assert: ships findAll size equals: 1.
+	ships
+		withOneMatching: [ :ship :criteria | criteria satisfyingAll: {( ship name = updatedSDF1 name )} ]
+		do: [ :ship | 
+			self assert: ship fleet size equals: 1.
+			ship fleet
+				do:
+					[ :member | updatedSDF1 fleet anySatisfy: [ :updatedSDF1Member | member name = updatedSDF1Member name ] ]
+			]
+		else: [ self fail ]
+]
+
+{ #category : #tests }
+PooledRDBMSRepositoryProviderTest >> testUpdateMothershipOriginallyWithEmptyFleetToNowHaveSingleShipFleet [
+
+	| sdf1 updatedSDF1 vf1A vf1D vf1J vf1S |
+
+	vf1A := Spaceship named: 'VF-1A'.
+	vf1D := Spaceship named: 'VF-1D'.
+	vf1J := Spaceship named: 'VF-1J'.
+	vf1S := Spaceship named: 'VF-1S'.
+	sdf1 := Mothership named: 'SDF-1' withAll: {}.
+	updatedSDF1 := Mothership
+		named: 'SDF-1*'
+		withAll:
+			{vf1A.
+			vf1D.
+			vf1J.
+			vf1S}.
+	ships store: sdf1.
+	ships update: sdf1 with: updatedSDF1.
+	self assert: ships findAll size equals: 1.
+	ships
+		withOneMatching: [ :ship :criteria | criteria satisfyingAll: {( ship name = updatedSDF1 name )} ]
+		do: [ :ship | 
+			self assert: ship fleet size equals: 4.
+			ship fleet
+				do:
+					[ :member | updatedSDF1 fleet anySatisfy: [ :updatedSDF1Member | member name = updatedSDF1Member name ] ]
+			]
+		else: [ self fail ]
+]
+
+{ #category : #tests }
+PooledRDBMSRepositoryProviderTest >> testUpdateMothershipOriginallyWithMultipleShipFleetToNowHaveEmptyFleet [
+
+	| sdf1 updatedSDF1 vf1A vf1D |
+
+	vf1A := Spaceship named: 'VF-1A'.
+	vf1D := Spaceship named: 'VF-1D'.
+	sdf1 := Mothership
+		named: 'SDF-1'
+		withAll:
+			{vf1A.
+			vf1D}.
+	updatedSDF1 := Mothership named: 'SDF-1*' withAll: {}.
+	ships store: sdf1.
+	ships update: sdf1 with: updatedSDF1.
+	self assert: ships findAll size equals: 1.
+	ships
+		withOneMatching: [ :ship :criteria | criteria satisfyingAll: {( ship name = updatedSDF1 name )} ]
+		do: [ :ship | self assert: ship fleet isEmpty ]
+		else: [ self fail ]
+]
+
+{ #category : #tests }
+PooledRDBMSRepositoryProviderTest >> testUpdateMothershipOriginallyWithMultipleShipFleetToNowHaveMultipleShipFleetWhichNoLongerIncludesTheOriginalShip [
+
+	| sdf1 updatedSDF1 vf1A vf1D vf1J vf1S |
+
+	vf1A := Spaceship named: 'VF-1A'.
+	vf1D := Spaceship named: 'VF-1D'.
+	vf1J := Spaceship named: 'VF-1J'.
+	vf1S := Spaceship named: 'VF-1S'.
+	sdf1 := Mothership
+		named: 'SDF-1'
+		withAll:
+			{vf1A.
+			vf1D}.
+	updatedSDF1 := Mothership
+		named: 'SDF-1*'
+		withAll:
+			{vf1J.
+			vf1S}.
+	ships store: sdf1.
+	ships update: sdf1 with: updatedSDF1.
+	self assert: ships findAll size equals: 1.
+	ships
+		withOneMatching: [ :ship :criteria | criteria satisfyingAll: {( ship name = updatedSDF1 name )} ]
```

Comment:
```suggestion
		withOneMatching: [ :ship | ship name = updatedSDF1 name  ]
```
BTW, Maybe you can extract these repeated code in a method named withShipNamed: ..